lib/Dav/AntivirusPlugin.php

```diff
@@ -90,31 +105,61 @@ public function beforeMove($source, $destination) {
 	 * @param INode $parentNode
 	 * @param bool $modified should be set to true, if this event handler
 	 *                           changed &$data
+	 * @return bool|null
+	 * @throws Forbidden
 	 */
 	public function beforeCreateFile($path, &$data, INode $parentNode, &$modified) {
-		$container = $this->application->getContainer();
-		$server = $container->getServer();
 		// Scan a public upload
-		if ($server->getUserSession()->getUser() === null) {
-			$appConfig = $container->query('AppConfig');
-			$scannerFactory = $container->query('ScannerFactory');
-			$scanner = $scannerFactory->getScanner();
-			$status = $scanner->scan(new Resource($data, $appConfig->getAvChunkSize()));
-			if (\intval($status->getNumericStatus()) === Status::SCANRESULT_INFECTED) {
-				$details = $status->getDetails();
-				$server->getLogger()->warning(
-					"Infected file deleted after uploading to the public folder. $details Path: $path",
-					['app' => 'files_antivirus']
-				);
-				throw new Forbidden(
-					$container->query('L10N')->t(
-						'Virus %s is detected in the file. Upload cannot be completed.',
-						$status->getDetails()
-					), false
-				);
-			}
+		if ($this->userSession->getUser() === null) {
+			$this->scanPublicUpload($path, $data);
+		}
+		\rewind($data);
+		return true;
+	}
+
+	/**
+	 * This method is triggered before a content written (but not for new files).
+	 *
+	 * @param string $path
+	 * @param INode $node
+	 * @param resource $data
+	 * @param bool $modified should be set to true, if this event handler
+	 *                           changed &$data
+	 * @return bool|null
+	 * @throws Forbidden
+	 */
+	public function beforeWriteContent($path, INode $node, &$data, &$modified) {
+		// Scan a public upload
+		if ($this->userSession->getUser() === null) {
```

Comment:
this check was not present before. Is this codepath really only for public uploads?

Reply:
the whole method `beforeWriteContent` is a brand new one. 
as for `beforeCreateFile`  method  - this check existed there: https://github.com/owncloud/files_antivirus/blob/3873ea322230fd0544e545839703ad5def7a2b0c/lib/Dav/AntivirusPlugin.php#L98